src/main/java/org/opensearch/geospatial/ip2geo/cache/GeoDataCache.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.ip2geo.cache;
+
+import java.net.InetAddress;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+
+import org.opensearch.common.cache.Cache;
+import org.opensearch.common.cache.CacheBuilder;
+
+/**
+ * Cache to hold geo data
+ */
+public class GeoDataCache {
+    private Cache<CacheKey, Map<String, Object>> cache;
+
+    public GeoDataCache(final long maxSize) {
```

Comment:
Shouldn't this be singleton?

Reply:
That make sense.

Reply:
Cannot make it as a singleton as the maxSize property is coming from the caller.

Reply:
Does it mean you have multiple cache per cluster?

Reply:
There will be single cache per node. Any suggestion on how you want to make it as singleton? I would leave the code as it is and let the user of this class to decide. Because same can happen for many other classes as well which are singleton but not protection on creating multiple instance.

Reply:
That is not true singleton; having single instance for each size. Also, there is no use case where we want single instance for each cache size.

Reply:
Other option is making this as a private inner class of another class in which this cache is used that I can make in a next PR.